### ISSUE TITLE
Doesn’t work for me

### ISSUE THREAD

**Issue by jirutka** [smasher164/pw-volume#2]
It behaves really weird, but I haven’t figured out what’s going on here yet. It looks like it doesn’t control the output volume but something else or there’s some weird conflict between controlling the output via `pw-cli` and via PulseAudio interface. It somehow messed the state and I had to reset Pipewire.

I’ll investigate it later, just wanted to let you know early.

- - -
Environment:
- Alpine Linux v3.15
- Pipewire 0.3.40

**Comment by smasher164** [comment]
I referenced this guide to translate pulseaudio commands to pipewire ones: https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Migrate-PulseAudio#sink-inputsource-output-volumemute.

So the gist of how I set the volume is
```
pw-cli s <node-id> Props '{ mute: <boolean>, channelVolumes: [ <float>, <float> ] }'
```
And everything before that is just to grab the node id. Does it work when you use `pw-cli` directly?

Also, when you say "weird conflict," do you mean that using `pw-volume` and `pactl` at the same time leads to some corrupted state? If so, I can install `pactl` to see if I can reproduce.

**Comment by mizzunet** [comment]
It seems it working improperly. Running `pw-volume change` plus increase or decrease any amount causes, the actual output volume stays at some lower amount that current.
Though, `pw-volume status` `percentage` changes accordingly. 

Steps:
1. Run `pipewire` and `wireplumber`
2. Play some audio
3. At this state, `pw-volume status` `percentage` is `25%`
4. `pw-volume change +10%`
5. Now, `percentage` is `35%`, but the output volume lowered than the current level
6. Changing again does nothing.

**Comment by mizzunet** [comment]
How do I find out `node-id` for `pw-cli s <node-id> Props '{ mute: <boolean>, channelVolumes: [ <float>, <float> ] }'`

**Comment by mizzunet** [comment]
I got it, node is `41`. I tried `pw-cli s 41 Props '{ mute: true}'` which mutes.
But `pw-cli s 41 Props '{ mute: false}'` unmutes and causes the distorted issue. So, I guess it's PipeWIre's issue

**Comment by smasher164** [comment]
What happens if you use 
```
pw-cli s 41 Props '{volume: <float>}'
```

Does the volume get set properly? If not, can you try it with `channelVolumes` instead?

```
pw-cli s 41 Props '{ channelVolumes: [ <float>, ... ]}'
```

If the latter works, I might end up reworking this command to set channelVolumes instead of the global volume field, the way pulseaudio does it.

**Comment by mizzunet** [comment]
```
pw-cli s 41 Props '{volume: 1.0}'
pw-cli s 41 Props '{ channelVolumes: [ 1.0 ]}'
```
Both do nothing

**Comment by mizzunet** [comment]
Both works fine

**Comment by mizzunet** [comment]
Wait, now `pw-volume change` also works fine. 

Let me try again after restaring

**Comment by mizzunet** [comment]
It seems working fine even after restarting. But there's one thing I noticed

**Comment by mizzunet** [comment]
`pw-volume` limits sound output. That is,

1. Starts `pipewire` and `wireplumber`
2. Plays some audio. Now `pw-volume status` is `{"percentage":10, "tooltip":"10%"}`
3. But `pw-volume change +10%` lowers much the actual output. Though `percentage` becomes `20`
4. Trying `pw-volume change +10%` util percentage reaches `100` but the ouput volume is way lower than sound at 2nd step

**Comment by mizzunet** [comment]
>  I'll follow up to see if y'all can test it.

I think, I can do that.

**Comment by smasher164** [comment]
I created PR https://github.com/smasher164/pw-volume/pull/10 to change `channelVolumes` like pactl, instead of `volume`. If you want to run it from source and test it on your system, let me know how it goes.

If that's not feasible, let me know and I can create a binary release.

**Comment by smasher164** [comment]
I've created a release containing these changes, and published corresponding binaries for x86_64 and aarch64.
https://github.com/smasher164/pw-volume/releases/tag/v0.2.0

If there is still an issue, please file a new, more specific one.
Thanks!

**Comment by jirutka** [comment]
I’m sorry for the late response. I’m testing it now. It seems that it works correctly now, but the volume change is somehow not propagated to the PulseAudio layer – the volume indicator in Waybar and PulseAudio Volume Control is not updated. That’s quite unexpected to me, but it’s probably some issue in Pipewire, not pw-volume.

**Comment by smasher164** [comment]
@jirutka What about the latest release? I think I might've figured out the root cause -- I was setting the stream volume, instead of the sink volume.
